firmware/mw_dual_motor_torque_ctrl/src/dual_motor_torque_ctrl.c

```diff
@@ -1171,34 +1196,53 @@ inline void generic_motor_ISR(const HAL_MtrSelect_e mtrNum)
 			speed_pu = STPOSCONV_getVelocity(st_obj[mtrNum].posConvHandle);
 		}
 		else
-		{  // the alignment procedure is in effect
-
-			// force motor angle and speed to 0
-			angle_pu[mtrNum] = _IQ(0.0);
-			speed_pu = _IQ(0.0);
-
-			// set D-axis current to Rs estimation current
-			gIdq_ref_pu[mtrNum].value[0] = _IQmpy(
+		{
+			// The alignment procedure is in effect.
+			// For the first gCalibrationWrampupTime steps, wramp up towards the
+			// calibration current slowly. Then, hold the current for
+			// the duration of gCalibrationHoldTime steps.
+			if(gAlignCount[mtrNum]++ < gCalibrationWrampupTime) {
+				angle_pu[mtrNum] = _IQ(0.0);
+				speed_pu = _IQ(0.0);
+
+				// Set D-axis current to slowly increasing estimation current.
+				_iq idq_ref_pu_max = _IQmpy(
+						_IQ(gUserParams[mtrNum].maxCurrent_resEst),
+						gCurrent_A_to_pu_sf[mtrNum]);
+				_iq idq_ref_pu_step = idq_ref_pu_max / gCalibrationWrampupTime;
+				gIdq_ref_pu_calibrate[mtrNum] += idq_ref_pu_step;
+				gIdq_ref_pu[mtrNum].value[0] = gIdq_ref_pu_calibrate[mtrNum];
+
+				// Set Q-axis current to 0.
+				gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
+			} else {
+				// force motor angle and speed to 0
+				angle_pu[mtrNum] = _IQ(0.0);
+				speed_pu = _IQ(0.0);
+
+				// set D-axis current to Rs estimation current
+				gIdq_ref_pu[mtrNum].value[0] = _IQmpy(
 					_IQ(gUserParams[mtrNum].maxCurrent_resEst),
-			        gCurrent_A_to_pu_sf[mtrNum]);
-			// set Q-axis current to 0
-			gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
+					gCurrent_A_to_pu_sf[mtrNum]);
+				// set Q-axis current to 0
+				gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
```

Comment:
There is a lot of duplication here with the "if"-case.  Maybe it could be restructured a bit to avoid this?

Reply:
Good point, I moved the 

```
            // force motor angle and speed to 0
            angle_pu[mtrNum] = _IQ(0.0);
            speed_pu = _IQ(0.0);

            // Set Q-axis current to 0.
            gIdq_ref_pu[mtrNum].value[1] = _IQ(0.0);
```

part before the if-statement as it's used by both if branches the same.